modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/request.ts

```diff
@@ -10,12 +9,180 @@
  * distribution rights of the Software.
  */
 
+import dayjs from 'dayjs';
+
+import {FIELDSREPORT, RequestFilterType} from '../../types';
 import fetcher from './fetcher';
+import {
+	filteredOrganizationsERC,
+	getIdGreaterOrEqualOrganization,
+	getIdLessOrEqualOrganization,
+	getOrganizationBetweenIds,
+	getOrganizationName,
+} from './organization';
 
 const resource = 'o/c/evprequests';
 
-export async function getRequestsByFilter() {
-	const response = await fetcher(`${resource}`);
+const nestedFields = '&nestedFields=r_organization_c_evpOrganization';
+
+const formatArrayUrl = (key: string, value: String[]) => {
+	let operator = `${key} in ({values})`;
+
+	operator = operator
+		.replace('{values}', value.map((value) => `'${value}'`).join(','))
+		.trim();
+
+	return operator;
+};
+
+const finalFormatRequestTime = (dateFinal: string) => {
+	return dateFinal.split('T')[0] + 'T23:59:59.999Z';
+};
+
+const createURLFilter = async (data: RequestFilterType) => {
+	let formatUrl;
+	let externalReferenceCodeOrganization;
+	const filterAnd = '/?filter=';
+	const filter = [];
+	let ISOFormattedInitialRequestDate;
+	let ISOFormattedFinalRequestDate;
+	let formattedTimeFinalRequestDate;
+
+	if (data.fullName) {
+		filter.push(`contains(${FIELDSREPORT.FULLNAME},'${data.fullName}')`);
+	}
+
+	if (data.requestStatus.length !== 0) {
+		formatUrl = formatArrayUrl(
+			FIELDSREPORT.REQUESTSTATUS,
+			data.requestStatus
+		);
+		filter.push(formatUrl);
+	}
+
+	if (data.liferayBranch.length !== 0) {
+		formatUrl = formatArrayUrl(
+			FIELDSREPORT.LIFERAYBRANCH,
+			data.liferayBranch
+		);
+		filter.push(formatUrl);
+	}
```

Comment:
There is the same behavior between these conditions. Please turn them into a generic function.

Reply:
const fuu = (x, y) => {
   formatUrl = formatArrayUrl(
			x,
			y
		);
    filter.push(formatUrl);
}